pkg/internal/polymorphichelpers/rollback.go

```diff
@@ -540,6 +543,20 @@ type AdvancedDaemonSetRollbacker struct {
 	kc kruiseclientsets.Interface
 }
 
+type RolloutRollbacker struct {
+	k  kubernetes.Interface
+	kc kruiseclientsets.Interface
+}
+
+// RolloutRollbacker.Rollback never be called
+func (r *RolloutRollbacker) Rollback(obj runtime.Object,
+	updatedAnnotations map[string]string,
+	toRevision int64,
+	dryRunStrategy cmdutil.DryRunStrategy) (string, error) {
+
```

Comment:
can we move the checkrollout logic here, and construct and call the corresponding rollback for workload?

Reply:
I've updated the code this week with the following enhancements:
1. Resources are now retrieved only once.
2. Added deduplication logic.
3. Enhanced error reporting for better clarity when undo operations fail.